skrub/_datetime_encoder.py

```diff
@@ -403,6 +422,12 @@ def _guess_datetime_format(X_col):
     # raises a TypeError.
     # We have to convert these to the object dtype first.
     X_col = X_col.astype("object")
+
+    # Subsample samples for fast format estimation
+    size = min(X_col.shape[0], 10)
```

Comment:
i guess we could expose the number of subsamples as a parameter. although if days are roughly uniformly distributed the probability of getting one < 12 10 times is very low so that's probably ok. we might want to give the constant a name or mention the value in the docstring